nixos/modules/services/backup/tarsnap.nix

```diff
@@ -329,7 +345,50 @@ in
           CapabilityBoundingSet = [ "CAP_DAC_READ_SEARCH" ];
           PermissionsStartOnly = "true";
         };
-      }) gcfg.archives;
+      }) gcfg.archives) //
+
+      (mapAttrs' (name: cfg: nameValuePair "tarsnap-restore-${name}"{
+        description = "Tarsnap restore '${name}'";
+        requires    = [ "network-online.target" ];
+
+        path = [ pkgs.iputils pkgs.tarsnap pkgs.utillinux ];
+
+        ##
+        preStart = ''
+          while ! ping -q -c 1 v1-0-0-server.tarsnap.com &> /dev/null; do sleep 3; done
```

Comment:
Nitpick: should probably be split to multiple lines to fit 80 chars.

Reply:
This sits at 88 characters, but only because of indentation. I think leaving it as-is is more readable than splitting it.